### PR TITLE
fix: yarn version not specified in monorepo

### LINKED ISSUE #7370: Repo: should specify yarn version
### Suggestion

The monorepo does not specify the package manager version in the package.json file, which is best practice AFAIK.

This is a problem because if you clone the repo and type `yarn`, it will automatically choose the latest version of Yarn on your system, and if you have Yarn 3 installed, then the monorepo installation will fail.

### PR DESCRIPTION
**Pull request by Zamiell**
## PR Checklist

- [X] Addresses an existing open issue: fixes #7370
- [ ] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [X] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

Adds explicit yarn version to prevent broken installs for new people cloning the repository.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -144,5 +144,6 @@
     "pretty-format": "^29",
     "tsx": "^3.12.7",
     "typescript": "5.1.6"
-  }
+  },
+  "packageManager": "yarn@1.22.19"
```
Comment (JoshuaKGoldberg) [inline package.json:148]: @JamesHenry I'll defer to you as #6162 is upcoming

Comment (Zamiell) [reply package.json:148]: that shouldn't really affect this pr

File: packages/website/blog/2023-07-09-announcing-typescript-eslint-v6.md [hunk 2]
```diff
@@ -63,7 +63,7 @@ As a result, we've reworked the configurations provided by typescript-eslint int
 
 - Functional rule configurations, for best best practices and code correctness:
   - **`recommended`**: Recommended rules that you can drop in without additional configuration.
-  - **`recommended-type-checked`**: Additional recommended rules that require type information.
+  - **`recommended-type-checked`**: Additional recommended rules that require type information.
```
Comment (JamesHenry) [inline packages/website/blog/2023-07-09-announcing-typescript-eslint-v6.md:66]: What's the deal with this diff?

Comment (Zamiell) [reply packages/website/blog/2023-07-09-announcing-typescript-eslint-v6.md:66]: no idea, it showed up as different when I copy pasted the md file from the main branch. are they different kind of spaces or something? I tried copy pasting them into [this site](https://www.babelstone.co.uk/Unicode) and they both show up as normal spaces, so I'm not really sure

**Comment by JamesHenry** [comment]
Now that #6162 is finally up to date and green, we can potentially skip this PR as the same field is added in that PR for yarn v3

**Comment by JoshuaKGoldberg** [comment]
Looks like #6162 is pretty close! I'll merge this in the meantime to give Zamiell contribution credit in the Git history / on github dot com. 😄

File: packages/website/blog/2023-07-09-announcing-typescript-eslint-v6.md (hunk 2 above)
Comment (JoshuaKGoldberg) [reply packages/website/blog/2023-07-09-announcing-typescript-eslint-v6.md]: `git checkout main -- packages/website/blog/2023-07-09-announcing-typescript-eslint-v6.md` fixed it for me. 🤷

**Comment by JoshuaKGoldberg** [comment]
Blurgh, I didn't think to change the PR title to `chore:`, and now it's in the release log. Ah well 🙃

**Pull request merged**